test/Test/Data/Mutable/Vector.hs

```diff
@@ -220,6 +238,79 @@ lenPushTest val vec = fromLen Linear.$ Vector.size vec
     fromLen (Ur len, vec) =
       compInts (move (len+1)) (getFst (Vector.size (Vector.push val vec)))
 
+refWrite :: Property
+refWrite = property $ do
+  l <- forAll nonEmptyList
+  ix <- forAll $ Gen.element [0..(length l - 1)]
+  v <- forAll val
+  let l' = listWrite ix v l
+  l' === unur (Vector.fromList l (Vector.toList Linear.. Vector.set ix v))
+  where
+
+    listWrite :: Int -> a -> [a] -> [a]
+    listWrite n _ _ | n Prelude.< 0 = error "Index negative"
+    listWrite _ _ [] = error "Index too big"
+    listWrite 0 a (_:xs) = a:xs
+    listWrite n a (x:xs) = x : listWrite (n-1) a xs
+
+refModify1 :: Property
+refModify1 = property $ do
+  l <- forAll nonEmptyList
+  let f x = (mod x 5, (mod x 5) Prelude.< 3)
+  ix <- forAll $ Gen.element [0..(length l - 1)]
+  let (_,result) = listApp ix f l
```

Comment:
You should do as in your test specification and write `snd (f (l !! ix))`.

Reply:
Forgot to do this in my last commit. Just force pushed now.